test/utilities/new_versions.jl

```diff
@@ -110,3 +110,234 @@ end
+    @test CompatHelper.continue_with_pr(pass_dep, true)
+
+    # Latest Version in Version Spec
+    dep = CompatHelper.DepInfo(
+        CompatHelper.Package("PackageA", UUID(1));
+        latest_version=VersionNumber(1),
+        version_spec=CompatHelper.VersionSpec(["0.9", "1.0"]),
+    )
+    @test !CompatHelper.continue_with_pr(dep, false)
+    @test !CompatHelper.continue_with_pr(dep, true)
+
+    # Equality Specifier
+    dep = CompatHelper.DepInfo(
+        CompatHelper.Package("PackageA", UUID(1));
+        version_verbatim="= 1.2",
+        latest_version=VersionNumber(2),
+    )
+    @test !CompatHelper.continue_with_pr(dep, false)
+    @test CompatHelper.continue_with_pr(dep, true)
+
+    # No Latest Version
+    dep = CompatHelper.DepInfo(CompatHelper.Package("PackageA", UUID(1)))
+    @test !CompatHelper.continue_with_pr(dep, false)
+    @test !CompatHelper.continue_with_pr(dep, true)
+end
+
+@testset "create_ssh_private_key" begin
+    mktempdir() do tmpdir
+        withenv(CompatHelper.PRIVATE_SSH_ENVVAR => "foo") do
+            pkey = apply(decode_pkey_patch) do
+                CompatHelper.create_ssh_private_key(tmpdir)
+            end
+
+            @test isfile(pkey)
+            open(pkey, "r") do io
+                @test read(io, String) == "pkey_info\n"
+            end
+        end
+    end
+end
+
+@testset "add_compat_entry" begin
+    mktempdir() do tmpdir
+        # Lets copy our test Project.toml to the tmpdir for this test
+        src = joinpath(@__DIR__, "..", "deps", "Project.toml")
+        dst = joinpath(tmpdir, "Project.toml")
+        cp(src, dst; force=true)
+
+        project = TOML.parsefile(dst)
+        @test !haskey(project["compat"], "PackageA")
+
+        CompatHelper.add_compat_entry("PackageA", tmpdir, "= 1.2")
+
+        project = TOML.parsefile(dst)
+        @test project["compat"]["PackageA"] == "= 1.2"
+    end
+end
+
+@testset "make_pr_for_new_version" begin
+    # Don't Continue with PR, `latest_version` is nothing
+    @test isnothing(
+        CompatHelper.make_pr_for_new_version(
+            GitHub.GitHubAPI(),
+            "hostname",
+            GitHub.Repo(),
+            CompatHelper.DepInfo(CompatHelper.Package("PackageA", UUID(1))),
+            CompatHelper.KeepEntry(),
+            CompatHelper.GitHubActions()
+        )
+    )
+
+    # return nothing: pr_title has already been created
+    apply(pr_titles_mock) do
+        @test isnothing(
+            CompatHelper.make_pr_for_new_version(
+                GitHub.GitHubAPI(),
+                "hostname",
+                GitHub.Repo(),
+                CompatHelper.DepInfo(
+                    CompatHelper.Package("PackageA", UUID(1));
+                    latest_version=VersionNumber(1),
+                    version_verbatim="0.9",
+                ),
+                CompatHelper.KeepEntry(),
+                CompatHelper.GitHubActions()
+            )
+        )
+    end
+
+    # Successful Run
```

Comment:
GitHub isn't allowing me to create this in its own testset, maybe too big of a change at once here.
The reason to be picky here is if tests are failing it is much more clear where in the test file it's failing as you know the `@testset` macro to search up.

Reply:
This is fair. I'm trying not to `over` testset everything, but there's a few places where it probably still makes sense to turn into a testset